x-pack/plugins/data_visualizer/public/application/common/components/document_count_content/document_count_content.tsx

```diff
@@ -210,9 +219,7 @@ export const DocumentCountContent: FC<Props> = ({
                             ? closestPrev
                             : closestNext;
 
-                        if (setSamplingProbability) {
-                          setSamplingProbability(closestProbability / 100);
-                        }
+                        updateSamplingProbability(closestProbability / 100);
```

Comment:
it would be more performant to put all of the logic in this `onChange` function inside the function that is wrapped in the debounce. There's no need to be calculating the `closestProbability` on every change if it's just going to be discarded.

Also there is a useful hook called `useDebounce` which might work well here.
You could put the `e.currentTarget.value` in a temporary state variable e.g. `newProbability` and then `useDebounce` could watch for changes in that var.